test/OpenTelemetry.Tests/Metrics/MeterProviderSdkTest.cs

```diff
@@ -96,13 +96,9 @@ public void TransientMeterExhaustsMetricStorageTest(bool withView, bool forceFlu
             Assert.Single(exportedItems);
         }
 
-#if DEBUG
-        // Note: This is inside a debug block because when running in CI the
-        // event source sees events from other tests running in parallel.
-        var metricInstrumentIgnoredEvents = inMemoryEventListener.Events.Where((e) => e.EventId == 33);
+        var metricInstrumentIgnoredEvents = inMemoryEventListener.Events.Where((e) => e.EventId == 33 && e.Payload[1] as string == meterName);
```

Comment:
@CodeBlanch FYI as you had added this unit test in #4958

Reply:
Oh nice you filtered it by the meter name 👍